src/server.py

```diff
@@ -0,0 +1,113 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from urllib.parse import urlparse, parse_qs
+from cowpy import cow
+import json
+import sys
+
+
+class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        parsed_path = urlparse(self.path)
+        parsed_qs = parse_qs(parsed_path.query)
+
+        # import pdb; pdb.set_trace()
+
+        if parsed_path.path == '/':
+            self.send_response(200)
+            self.end_headers()
+
+            self.wfile.write(return_html_string())
+            return
+
+        elif parsed_path.path == '/cowsay':
+            self.send_response(200)
+            self.end_headers()
+
+            self.wfile.write(b'Helpful instructions about this application')
+            return
+
+        
+        elif parsed_path.path == '/cow':
+            try:
+                # import pdb; pdb.set_trace()
+                msg = parsed_qs['msg'][0]
+                print(msg)
+            except (KeyError, json.decoder.JSONDecodeError):
+                self.send_response(400)
+                self.end_headers()
+                self.wfile.write(b'You did a bad thing')
+                return
+
+            cheese = cow.Moose(thoughts=True)
+            message = cheese.milk(msg)
+
+            self.send_response(200)
+            self.end_headers()
+            self.wfile.write(message.encode('utf8'))
+            return
+
+        else:
+            self.send_response(404)
+            self.end_headers()
+            self.wfile.write(b'Not Found')
+
+    def do_POST(self):
+        parsed_path = urlparse(self.path)
+        parsed_qs = parse_qs(parsed_path.query)
+        
+        if parsed_path.path == '/cow':
+            try:
+                msg = parsed_qs['msg'][0]
```

Comment:
The data for a POST request is not stored in the query string. It is in a file attached to the request.